qr_code/qrcode/utils.py

```diff
@@ -1,96 +1,198 @@
+        :param alignment_light: Color of the light alignment modules (default: same as ``light``)
+        :param timing_dark: Color of the dark timing pattern modules (default: same as ``dark``)
+        :param timing_light: Color of the light timing pattern modules (default: same as ``light``)
+        :param separator: Color of the separator (default: same as ``light``)
+        :param dark_module: Color of the dark module (default: same as ``dark``)
+        :param quiet_zone: Color of the quiet zone modules (default: same as ``light``)
+
+        The *size* parameter gives the size of each module of the QR code matrix. It can be either a positive integer or one of the following letters:
+            * t or T: tiny (value: 6)
+            * s or S: small (value: 12)
+            * m or M: medium (value: 18)
+            * l or L: large (value: 30)
+            * h or H: huge (value: 48)
+
+        For PNG image format the size unit is in pixels, while the unit is 0.1 mm for SVG format.
+
+        The *border* parameter controls how many modules thick the border should be (blank space around the code).
+        The default is 4, which is the minimum according to the specs.
+
+        The *version* parameter is an integer from 1 to 40 that controls the size of the QR code matrix. Set to None to determine
+        this automatically. The smallest, version 1, is a 21 x 21 matrix. The biggest, version 40, is 177 x 177 matrix.
+        The size grows by 4 modules/side.
+
+        There are 4 error correction levels used for QR codes, with each one adding different amounts of "backup" data
+        depending on how much damage the QR code is expected to suffer in its intended environment, and hence how much
+        error correction may be required. The correction level can be configured with the *error_correction* parameter as follow:
+            * l or L: error correction level L – up to 7% damage
+            * m or M: error correction level M – up to 15% damage
+            * q or Q: error correction level Q – up to 25% damage
+            * h or H: error correction level H – up to 30% damage
+        :raises: TypeError in case an unknown argument is given.
+        """
+        self._size = size
+        self._border = border
+        if _can_be_cast_to_int(version):
+            version = int(version)
+            if not 1 <= version <= 40:
+                version = None
+        elif version in ('m1', 'm2', 'm3', 'm4', 'M1', 'M2', 'M3', 'M4'):
+            version = version.lower()
+            # Set / change the micro setting otherwise Segno complains about
+            # conflicting parameters
+            micro = True
+        else:
+            version = None
+        self._version = version
+        if not isinstance(micro, bool):
+            micro = micro == 'True'
+        self._micro = micro
+        try:
+            error = error_correction.lower()
```

Comment:
Since option is called `error_correction`, shouldn't we call the internal property `error_correction` as well?